docs/flow/config-language/components.md

```diff
@@ -6,3 +6,96 @@ weight: 300
 ---
 
 # Components
+Components are the defining feature of Grafana Agent Flow. They are autonomous
+pieces of business logic that perform a single task (like retrieving secrets or
+collecting Prometheus metrics) and can be wired together to form programmable
+pipelines of telemetry data.
+
+Under the hood, components are orchestrated via the [_component
+controller_]({{< relref "../concepts/component_controller.md" >}}), who is
+responsible for scheduling them, reporting their health and debug status, and
+continuously re-evaluating their inputs and outputs.
+
+
+## Configuring components
+Each top-level River _block_ will instantiate a new component. All components
+are identified by their name, describing what the component is responsible for,
+while some allow or require to provide an extra user-specified _label_.
+
+You can see a list of all available components [here]({{< relref "../components/_index.md" >}}).
+Each one features a complete reference page, so getting a component to work for
+you should be as easy as reading its documentation and copy/pasting from an
+example.
```

Comment:
I vaguely remember seeing a guideline that recommended avoiding using "see here" type links, though I can't remember from where.

Can we rephrase to something like 

```
The [components reference](LINK) shows a list of all defined components. Each component has a compete reference page, ...  
```